collector/lib/runtime-control/Config.h

```diff
@@ -0,0 +1,54 @@
+#ifndef _RUNTIME_CONTROL_CONFIG_H_
+#define _RUNTIME_CONTROL_CONFIG_H_
+
+#include <chrono>
+#include <condition_variable>
+#include <mutex>
+#include <optional>
+
+#include <storage/runtime_filters.pb.h>
+
+#include "Hash.h"
+
+#include "../ResourceSelector.h"
+
+namespace collector::runtime_control {
+
+class Config {
+ public:
+  Config() {
+    for (int i = 0; i < storage::RuntimeFilterFeatures_ARRAYSIZE; i++) {
+      kFeatureFlags_[i] = 1UL << (i + 1);
+    }
+  }
+
+  static Config& GetOrCreate();
```

Comment:
[nitpick] If we want `Config` to be a singleton, the "traditional" naming for this is `GetInstance` I believe.
```suggestion
  static Config& GetInstance();
```

Reply:
You also want to make the default constructor private and `delete` all other default methods of construction so that we can't have multiple instances of the configuration floating around.

Reply:
Done